source/common/runtime/runtime_impl.cc

```diff
@@ -26,32 +23,57 @@ namespace Runtime {
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
 std::string RandomGeneratorImpl::uuid() {
-#ifdef __APPLE__
-  // TODO(zuercher): factor out a sensible interface for UUID generation to make it easier
-  // to put a faster implementation in later.
-  char generated_uuid[UUID_LENGTH + 1];
-  uuid_t uuid;
-  uuid_generate_random(uuid);
-  uuid_unparse(uuid, generated_uuid);
-  return std::string(generated_uuid);
-#else
-  int fd = open("/proc/sys/kernel/random/uuid", O_RDONLY);
-  if (-1 == fd) {
-    throw EnvoyException(fmt::format("unable to open uuid, errno: {}", strerror(errno)));
+  // Create UUID from Truly Random or Pseudo-Random Numbers.
+  // See: https://tools.ietf.org/html/rfc4122#section-4.4
+  uint8_t rand[16];
+  RAND_bytes(rand, 16);
```

Comment:
I'd suggest an `ASSERT` or `RELEASE_ASSERT` on the value, since it feels like we're relying on implementation detail that might conceivably change. Up to you which one.